### ISSUE TITLE
Feature Request: Multiple Included or Excluded Scopes

### ISSUE THREAD

**Issue by srosenberg-coveo** [vale-cli/vale#259]
I'm using Vale on Mac OS, installed via Homebrew, version 2.5.1. I know you already had an issue for multiple scopes (https://github.com/errata-ai/vale/issues/74), which you closed. However, we have an example that I think makes a clear case for this feature.

We have the following rule, which is meant to enforce capitalization for our product, component, and team names:

```
extends: substitution
message: "Use '%s' instead of '%s'."
ignorecase: false
level: error
action:
  name: replace
swap:
  ...
  "[wW]eb [sS]ource": Web source
  "[wW]eb [sS]ources": Web sources
```
We explicitly don't want this rule to operate in headings or table headers, because we use title case for them, but we do want it to operate in list items, table cells, and paragraphs/sentences. We could solve this by making three separate rules, one for each scope, but that kind of duplication invites a lot of opportunity for error.

It would be great if we could specify multiple scopes to include or exclude. Please let me know if you need more context, or if there's an alternative solution that I'm missing.

**Comment by jdkato** [comment]
My intent with #74 wasn't to imply that I'm not open to supporting multiple scopes&mdash;it's something that I'd like to do eventually.

However, I think the case you describe is somewhat usual&mdash;Vale's current approach to such a scenario would be:

1. Create a single, text-scoped `substitution` rule to enforce your product, component, and team names; and then

2. Add these terms as an exception to any `capitalization` rules, allowing your branding to remain consistent regardless of its scope.

**Comment by srosenberg-coveo** [comment]
I might be wrong, but I think that solves the opposite issue from ours. I'll try to explain more clearly using the above example. We have a component called a `Web source`, and we want to enforce that exact casing whenever it appears in context that **isn't** a heading or table header.

Right now we have a text-scoped `substitution` rule (I've extracted the relevant lines in the original post), and a standard `capitalization` rule for headings:

```
extends: capitalization
message: "Be sure to use title case!"
scope: heading
level: warning
match: $title
style: AP
```

We don't want names like `Web source` to be an exception to our heading capitalization rules - we want all of our headings to always be in title case (i.e., the heading `## Creating Web Sources` is correct). Elsewhere in our docs, though, we want to be sure that it's always written `Web source`. In the example below, which I already included in a separate bug report, I've written it correctly in the headings and incorrectly in the paragraph and list text, to test the rules:

![shorter-web-source-snippet](https://user-images.githubusercontent.com/54185735/98292021-06790e00-1f7a-11eb-95c7-cbbbb7e1d03d.png)

Note that the errors that are being flagged in the heading lines are not related to the capitalization rule, because the headings are correctly capitalized. But I currently have no way to prevent the substitution rule from flagging them as errors unless I create individual rules to cover all of the use cases in which I want our special casing rules to apply (paragraph/sentence, list, table cell, etc.)

**Comment by jdkato** [comment]
> I might be wrong, but I think that solves the opposite issue from ours. 

Yes, I was simply pointing out that Vale is currently designed to handle the problem from that (opposite) perspective. 

I think it's unusual to change your branding based on where it appears in a document (vs. keeping the branding consistent and adding an exception to the `capitalization` rule&mdash;i.e., you probably wouldn't write "iPhone" as "IPhone" to appease a `capitalization` rule).

In any case, this issue is on my TODO list.

**Comment by stalsma** [comment]
I wanted to add an additional use-case to this request.

My style guide requires acronyms to be defined (spelled out) the first time they are used; simple enough. However, this should not be enforced in headings, as it compromises heading readability.

As I understand it, I need to essentially (re-)define the same acronym style for every scope I want it active.  And even then, I suspect that it will be problematic; if the acronym is defined in the first paragraph, and then reused in a list, the list version will throw and error, as the acronym was not previously defined in the list scope.

A contrived example:

# Lesson Overview of ICD-10-PCS

In this lesson, we will learn about the structure of the International Classification of Diseases, 10th Revision, Procedure Coding System (ICD-10-PCS).

## Lesson Objective

Completing this lesson informs you how to 

* Recognize the structure of ICD-10-PCS

**Comment by plaindocs** [comment]
Just one extra use case here, I currently have a different style for H1 and for H2-7, as it stands I have to dupe for the rule for each heading level, right? instead of using `scope: heading,h.h2, heading.h3`...

**Comment by jdkato** [comment]
@plaindocs: That's correct, currently.

Work on this issue is underway though.

**Comment by jdkato** [comment]
This is available now in [v2.10.0](https://github.com/errata-ai/vale/releases/tag/v2.10.0).
